src/libs/actions/Report.js

```diff
@@ -2002,6 +2002,9 @@ function openReportFromDeepLink(url, isAuthenticated) {
         Session.waitForUserSignIn().then(() => {
             Navigation.waitForProtectedRoutes().then(() => {
                 const route = ReportUtils.getRouteFromLink(url);
+                if (!route) {
+                    return;
+                }
```

Comment:
It redirects after nav state hydration has already occurred here in the default report setter:
https://github.com/Expensify/App/blob/04b5fa452fb97579e6613033340c58c572aed8a0/src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js#L85-L91

I'm finding that just fixing the onyx key still leaves a flash that's too fast to catch on video, but it's observable if you try to repro with this branch. It isn't the composer size, so it's technically out of scope for this issue - happy to remove if you prefer. But resetting nav state twice, which only happens on wide screens loading the root domain with no path, is causing problems, we really shouldn't be doing it. It's covered in step 4 of the root cause in my proposal.

Reply:
The problem is a lot of UI conditionals hinge on whether there's a report id. When you load the root domain, the default report setter provides a report id, but then the deep link handler redirects, blowing up nav state and taking report id back to an empty string. This leaves a gap where various components are rendering when there's no report ID. We're counting on a bunch of defaults to cover this case, but there's no need for the double redirect in the first place.

Reply:
Alright, I'll update the PR.

Reply:
I appreciate you digging into this and trying to fix the problem. I agree that it would be great to handle that in a separately scoped issue and PR. That makes everything easier to review/test/revert.